src/plugin.js

```diff
@@ -209,6 +216,9 @@ const contribAdsPlugin = function(options) {
   // Start sending contentupdate events for this player
   initializeContentupdate(player);
 
+  // Global contentupdate handler for resetting plugin state
+  player.on('contentupdate', player.ads.initialize);
```

Comment:
Since this is a recurring event,  renaming `player.ads.initialize` to `player.ads.reset` feels more natural to me